src/components/picker/PickerItem.js

```diff
@@ -16,153 +16,138 @@ import Text from '../text';
-class PickerItem extends BaseComponent {
-  static displayName = 'Picker.Item';
-
-  static propTypes = {
-    /**
-     * Item's value
-     */
-    value: PropTypes.oneOfType([PropTypes.object, PropTypes.string, PropTypes.number]),
-    /**
-     * Item's label
-     */
-    label: PropTypes.string,
-    /**
-     * Custom function for the item label (e.g (value) => customLabel)
-     */
-    getItemLabel: PropTypes.func,
-    /**
-     * DEPRECATE: Function to return the value out of the item value prop when value is custom shaped.
-     */
-    getItemValue: PropTypes.func,
-    /**
-     * Is the item selected
-     */
-    isSelected: PropTypes.bool,
-    /**
-     * Pass to change the selected icon
-     */
-    selectedIcon: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
-    /**
-     * Pass to change the selected icon color
-     */
-    selectedIconColor: PropTypes.string,
-    /**
-     * Is the item disabled
-     */
-    disabled: PropTypes.bool,
-    /**
-     * Render custom item
-     */
-    renderItem: PropTypes.elementType,
-    /**
-     * Callback for onPress action
-     */
-    onPress: PropTypes.func,
-    /**
-     * Callback for onLayout event
-     */
-    onSelectedLayout: PropTypes.func
-  };
-
-  
-  constructor(props) {
-    super(props);
-
-    if (_.isPlainObject(props.value)) {
+const PickerItem = props => {
+  const {
+    renderItem,
+    value,
+    label,
+    onPress,
+    disabled,
+    isSelected,
+    selectedIcon = Assets.icons.check,
+    selectedIconColor,
+    testID
+  } = props;
+
+  useEffect(() => {
+    if (_.isPlainObject(value)) {
       console.warn('UILib Picker.Item will stop supporting passing object as value & label (e.g {value, label}) in the next major version. Please pass separate label and value props');
     }
-  }
-  
+  }, []);
 
-  generateStyles() {
-    this.styles = createStyles(this.props);
-  }
+  // TODO: deprecate the check for object
+  const _onPress = useCallback(() => {
+    // onPress(_.isObject(value) ? value : {value, label});
+    onPress(value);
+  }, [value, onPress]);
 
-  getLabel() {
-    const {value, label} = this.props;
+  const onSelectedLayout = useCallback((...args) => {
+    _.invoke(props, 'onSelectedLayout', ...args);
+  }, []);
 
+  const getLabel = () => {
     if (_.isObject(value)) {
-      return _.invoke(this.props, 'getItemLabel', value) || _.get(value, 'label');
+      return _.invoke(props, 'getItemLabel', value) || _.get(value, 'label');
     }
     return label;
-  }
-
-  renderSelectedIndicator() {
-    const {
-      isSelected,
-      disabled,
-      selectedIcon = Assets.icons.check,
-      selectedIconColor = Colors.primary
-    } = this.props;
+  };
 
+  const renderSelectedIndicator = () => {
     if (isSelected) {
       return <Image source={selectedIcon} tintColor={disabled ? Colors.dark60 : selectedIconColor}/>;
     }
-  }
-
-  renderItem() {
-    const {disabled} = this.props;
+  };
 
+  const _renderItem = () => {
```

Comment:
I don't want to overkill the component with memoization.
I read in a blog post about when to useMemo and it says it's not recommended to overuse memoization cause it the it can also affect performance. 
From what I read, the thumb rule is to start simple and to consider memoizing when we face performance issues. 

memoizing this will require wrapping `getLabel` with `useCallback` and I prefer not to complicate it at the moment. 

Anyway, Personally I still not certain what is best practice with how often to use `useMemo` so again, for now I prefer to go easy on it..

Reply:
Gotcha. confusing topic indeed :)